thirdparty/src/main/java/org/cache2k/benchmark/thirdparty/TCache1Factory.java

```diff
@@ -0,0 +1,102 @@
+/*********************************************************************************
+ * Copyright 2016-present trivago GmbH
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **********************************************************************************/
+
+package org.cache2k.benchmark.thirdparty;
+
+import com.trivago.triava.tcache.TCacheFactory;
+import com.trivago.triava.tcache.core.Builder;
+import com.trivago.triava.tcache.eviction.Cache;
+import org.cache2k.benchmark.BenchmarkCache;
+import org.cache2k.benchmark.BenchmarkCacheFactory;
+
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.concurrent.TimeUnit.*;
+
+/**
+ * Factory for TCache
+ *
+ * @author Christian Esken
+ */
+public class TCache1Factory extends BenchmarkCacheFactory {
+
+  @Override
+  public BenchmarkCache<Integer, Integer> create(int _maxElements) {
+    MyBenchmarkCacheAdapter c = new MyBenchmarkCacheAdapter();
+    c.size = _maxElements;
+    createCache(_maxElements, c);
+    return c;
+  }
+
+  private void createCache(final int _maxElements, final MyBenchmarkCacheAdapter _adapter) {
+	
+	  Builder<Integer, Integer> b = TCacheFactory.standardFactory().builder();
+	  b.setExpectedMapSize(_maxElements);
+
+      /**
+       * Set a cache name. For some reason, auto-assigning a name does not work properly in this JMH based test.
+       * The issue is that the TCacheFactory.anonymousCacheId.incrementAndGet() returns 1 in all cases. This
+       * makes no real sense, as that AtomicInteger variable is static. It may be related to JVM forking by JMH, but
+       * it still seems odd. For now, just generate random Cache names, in the hope they will not clash.
+       */
+    String id = "tcache-" + new Random().nextInt();
```

Comment:
I would prefer just a static name, e.g. "benchmarkCache". There should be only one cache instantiated in the benchmarks, otherwise it needs to be fixed.